### PR TITLE
Improve logs component

### PR DESCRIPTION
**Pull request by laulauland**
resolves FP-4114, FP-4116, FP-4118, FP-4128

This PR introduces some improvements to the logs component based on the new designs. The component is now a little more responsive, uses `<details><summary>` HTML components for managing expanded state, features the timestamp in the appropriate spot, and adds a button for quickly copying the message to clipboard. When there are error-level logs, the logs panel gets expanded automatically and also shows a red dot on the bottom-right corner icon

Besides that the PR:
- normalizes the spacing of the overall layout.
- refactors some of the layout-level components to make them more readable.
- adds some code to attempt at creating a concurrency bug in the goose-quotes sample app.

### CODE REVIEW

File: studio/src/pages/RequestorPage/LogsTable/LogsTable.tsx [hunk 1]
```diff
@@ -119,104 +46,142 @@ export function LogsTable({ traceId = "" }: Props) {
...
-  data,
-  expandedRowId,
-  setExpandedRowId,
-}: TableProps) {
-  const table = useReactTable({
-    data,
-    columns,
-    getCoreRowModel: getCoreRowModel(),
-  });
+function LogsGrid({ logs }: LogsGridProps) {
```
Comment (flenter) [inline studio/src/pages/RequestorPage/LogsTable/LogsTable.tsx]: Now the code is this simple, i would merge this component with the LogsTable component.

File: studio/src/pages/RequestorPage/LogsTable/LogsTable.tsx [hunk 2]
```diff
@@ -119,104 +46,142 @@ export function LogsTable({ traceId = "" }: Props) {
...
+      ))}
+    </div>
+  );
+}
+
+type LogRowProps = {
+  log: MizuOrphanLog;
+};
+
+function LogRow({ log }: LogRowProps) {
```
Comment (flenter) [inline studio/src/pages/RequestorPage/LogsTable/LogsTable.tsx:67]: Bonus points if this file can be split into multiple files.

File: studio/src/Layout/BottomBar.tsx
Comment (flenter) [inline studio/src/Layout/BottomBar.tsx:1]: Oh in a separate file! ❤️

File: studio/src/components/IconWithNotification.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+const IconWithNotification: React.FC<IconWithNotificationProps> = ({
+  notificationColor = "bg-red-500",
+  notificationSize = 10,
+  notificationPosition = "top-right",
+  notificationContent,
+  showNotification = true,
+  ...iconProps
+}) => {
+  const getNotificationStyle = (): React.CSSProperties => {
+    const baseStyle: React.CSSProperties = {
```
Comment (flenter) [inline studio/src/components/IconWithNotification.tsx]: Makes me wonder why not use tailwind (for the more typical things like flex/asbsolute/alignment etc)

**Review by flenter — APPROVED** [review]

**Review by mies — APPROVED** [review]

File: studio/src/pages/RequestorPage/LogsTable/LogsTable.tsx (hunk 2 above)
Comment (laulauland) [reply studio/src/pages/RequestorPage/LogsTable/LogsTable.tsx:67]: It's small enough I kinda want to keep it in one for now

**Pull request merged**